glyphs-reader/src/font.rs

```diff
@@ -264,3 +294,296 @@ impl Path {
+                            return Err(Error::StructuralError("No value for Axes custom parameter".into()));
+                        };
+                        for v2_axis in v2_axes {
+                            let Plist::Dictionary(v2_axis) = v2_axis else {
+                                return Err(Error::StructuralError("Axis value must be a dict".into()));
+                            };
+                            let tag = v2_axis.get("Tag").unwrap().as_str().unwrap();
+                            tags.push(tag.to_string()); 
+                            v3_axes.push(Axis {
+                                name: v2_axis.get("Name").unwrap().as_str().unwrap().into(),
+                                tag: tag.into(),
+                            });
+                        }
+                    }
+                }
+            }
+            axes_idx.map(|idx| custom_params.remove(idx));
+
+            if custom_params.is_empty() {
+                self.other_stuff.remove("customParameters");
+            }            
+        }
+        Ok(tags)
+    }
+
+    fn v2_to_v3_metrics(&mut self) -> Result<(), Error> {
+        // metrics are in parallel arrays in v3
+        let v3_metric_names = ["ascender".to_string(), "baseline".to_string(), "descender".to_string(), "cap height".to_string(), "x-height".to_string()];
+        let v2_metric_names = ["ascender".to_string(), "baseline".to_string(), "descender".to_string(), "capHeight".to_string(), "xHeight".to_string()];
+        assert!(v2_metric_names.len() == v3_metric_names.len());
+
+        // setup root storage for the basic metrics
+        let mut metrics = Vec::new();
+        for name in v3_metric_names {
+            metrics.push(Plist::Dictionary(BTreeMap::from([("type".to_string(), Plist::String(name.clone()))])));
+        }        
+        self.other_stuff.insert("metrics".into(), Plist::Array(metrics));
+
+        // in each font master setup the parallel array
+        for master in self.font_master.iter_mut() {
+            let mut metric_values = Vec::new();
+            for v2_name in v2_metric_names.iter() {
+                let mut dict = BTreeMap::new();
+                if let Some(Plist::Integer(value)) = master.other_stuff.remove(v2_name) {
+                    // leave blank for 0
+                    if value != 0 {
+                        dict.insert("pos".to_string(), Plist::Integer(value));
+                    }                    
+                }
+                metric_values.push(Plist::Dictionary(dict));
+            }
+            master.other_stuff.insert("metricValues".into(), Plist::Array(metric_values));
+        }
+        Ok(())
+    }
+
+    fn v2_to_v3_shapes(&mut self) -> Result<(), Error> {
+        // shapes in v3 encompasses paths and components in v2
+        for glyph in self.glyphs.iter_mut() {
+            // v2 uses single codepoint strings, turn into int to match v3 for now
```

Comment:
> v2 uses single codepoint strings

actually, a glyph's `unicode` attribute in v2 can contain a comma-separated list of hexadecimal unicode codepoints, not just one.

And sometimes Georg insisted in not adding quotes to disambiguate these hex strings from equivalent decimal digits so our openstep_plist parser would give as ints but what they really where was hex strings so we had to work around that...